driver-sync/src/test/functional/com/mongodb/client/csot/AbstractClientSideOperationsEncryptionTimeoutProseTest.java

```diff
@@ -200,12 +199,21 @@ void shouldThrowOperationTimeoutExceptionWhenDecryptData() {
      * Not a prose spec test. However, it is additional test case for better coverage.
      */
     @Test
-    @Disabled
-    // TODO JAVA-5322. We have to reset timeoutMS for encrypted command.
+    @Tag("setsFailPoint")
```

Comment:
This was just a pattern adopted from other code sources and allows for specific tests to manage turning off failpoints in the clean up.  Its preferred over extensions as its  done on a per test basis rather than the whole test class.

Reply:
> This was just a pattern adopted from other code sources

As far as I can see, this is a pattern that was introduced in 2024 in CSOT, it does not exist in `master`.

> its done on a per test basis rather than the whole test class

The same can be achieved by raising a flag being part of test's instance state, and turning off failpoints in the `@AfterEach` method based on that flag.

> It has the benefit, should we run our test suite in the future against a topology that doesn't have perms to set fail points, we can easily filter the tests out.

Everywhere in the driver we skip tests depending on the environment by using `org.junit.jupiter.api.Assumptions`/`org.junit.Assume`, which means that the `@Tag` approach is inconsistent with the current approach and it's unlikely that we will start skipping tests depending on the environment using another mechanism.

Thus, I still don't think that it's a good idea to use `@Tag`s for turning failpoints off.

P.S. I realize that this PR is not a place to make any changes with regard to the approach, even if we eventually agree that it shouldn't be used. So I'll leave a comment in https://github.com/mongodb/mongo-java-driver/pull/1215.

Reply:
The comment in the CSOT PR: https://github.com/mongodb/mongo-java-driver/pull/1215#discussion_r1536885331.